ccache.c

```diff
@@ -2441,6 +2441,31 @@ cc_process_args(struct args *args, struct args **preprocessor_args,
 			free(option);
 			continue;
 		}
+                if (str_startswith(argv[i], "-B")) {
+			char *relpath = make_relative_path(x_strdup(argv[i] + 2));
+			char *option = format("-B%s", relpath);
+			args_add(stripped_args, option);
+			free(relpath);
+			free(option);
+			continue;
+                }
+                if (str_startswith(argv[i], "-iprefix")) {
+			char *relpath = make_relative_path(x_strdup(argv[i+1]));
+			char *option = format("-iprefix %s", relpath);
+			args_add(stripped_args, option);
+			free(relpath);
+			free(option);
+			i++;
+			continue;
+                }
+                if (str_startswith(argv[i], "-L")) {
```

Comment:
Please reuse code between the `-B` and `-L` cases since they are identical except for `B` and `L`.